### PR TITLE
ZCS-1177: Universal UI- New / edit distribution list, header and members view

### PR DESCRIPTION
**Pull request by kalpeshhpatel**
Changes:

* Added/updated required CSS for `dwt.css`, `zm.css`, `skin.css`
* Added/updated required strings for `ZmMsg.properties` and `skin.properties`
* Added/updated templates as per requirement for `Contacts.template` and `Widget.template`
* ZmContact,js:
  * Added event property when contact's avatar changes
* ZmContactsApp.js
  * Removed icon for create DL operation
* ZmContactSplitView.js
  * Updated list view template to render contact's avatar if present
  * Updated `_modifyContact` event to handle only contact avatar change event
* ZmGroupView.js:
  * Removed absolute positioning for DL properties tab

https://jira.corp.synacor.com/browse/ZCS-1177

### CODE REVIEW

File: WebRoot/js/zimbraMail/abook/view/ZmContactSplitView.js [hunk 1]
```diff
@@ -1133,7 +1133,8 @@ ZmContactSimpleView.prototype._modifyContact =
 function(ev) {
 	ZmContactsBaseView.prototype._modifyContact.call(this, ev);
 
-	if (ev.getDetail("fileAsChanged")) {
+	//TODO: `contactImageChanged` event should be handled separately and should just update current row instead of relayout
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/abook/view/ZmContactSplitView.js]: can we file a performance improvement ticket for this, just to make sure we don't forget this in future

File: WebRoot/js/zimbraMail/abook/view/ZmContactSplitView.js [hunk 2]
```diff
@@ -1207,7 +1208,12 @@ function(contact, params, asHtml, count) {
 	}
 
 	// icon
-	htmlArr[idx++] = AjxImg.getImageHtml(contact.getIcon(folder), null, "id=" + this._getFieldId(contact, "type"),null, null, ["ZmContactIcon"]);
+	var avatarUrl = contact.getImageUrl();
+	if(avatarUrl) {
+		htmlArr[idx++] = "<div class='ZmContactIcon' id='" + this._getFieldId(contact, "type") + "' ><img src='" + avatarUrl + "' alt='" + contact.getFullName() + "' /></div>";
```
Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/abook/view/ZmContactSplitView.js]: small: We should avoid using string concatenation if possible. Maybe use htmlArr.push ( "string1" , "string2" ) something like that .

File: WebRoot/js/zimbraMail/abook/view/ZmGroupView.js [hunk 3]
```diff
@@ -554,7 +554,7 @@ function(x, y, width, height) {
 	if(this._addNewField){
 		Dwt.setSize(this._addNewField, Dwt.DEFAULT, 100);
 	}
-	var memberListHeight = height - 50;
+	var memberListHeight = height - 150;
```
Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/abook/view/ZmGroupView.js]: can we add a comment what determines the subtract value . Could be helpful for someone looking this in future.

File: WebRoot/skins/_base/base4/skin.css [hunk 4]
```diff
@@ -997,3 +997,33 @@ div.ZmHtmlEditor .TxtAreaDrag {
...
+.DwtChooserListViewCustomStyle .DwtChooserListView .DwtListView-Column {
+    @ChooserListViewCustomStyleColumn@
+}
+
+.DwtChooserListViewCustomStyle .DwtChooserListView .DwtListView-Rows {
+    @ChooserListViewCustomStyleContainer@
+}
+
+/* For DL edit, hide location */
+.ZmDlViewWrapper .ZmContactHeaderLocation {
```
Comment (silentsakky) [inline WebRoot/skins/_base/base4/skin.css]: should this be moved to zm.css as it looks like part of functionality

File: WebRoot/skins/_base/base4/skin.css [hunk 5]
```diff
@@ -997,3 +997,33 @@ div.ZmHtmlEditor .TxtAreaDrag {
...
+    display: none;
```
Comment (tarangkhandelwal) [inline WebRoot/skins/_base/base4/skin.css]: can we do it from template somehow?

File: WebRoot/css/zm.css [hunk 6]
```diff
@@ -6703,6 +6698,44 @@ a.ZmLinkDisabled {
...
+	@DLEditViewWrapper@
+}
+
+.ZmDlViewWrapper .ZmContactHeader {
+	@DLEditViewHeader@
+}
+
+.ZmDlViewWrapper .contactHeader2,
+.ZmDlViewWrapper .contactHeader2 .Text {
+	@GroupEditViewInput@
```
Comment (silentsakky) [inline WebRoot/css/zm.css]: can we use 
DLEditViewInput = GroupEditViewInput

File: WebRoot/js/zimbraMail/abook/view/ZmContactSplitView.js (hunk 1 above)
Comment (kalpeshhpatel) [inline WebRoot/js/zimbraMail/abook/view/ZmContactSplitView.js]: Done. 

Thanks

File: WebRoot/js/zimbraMail/abook/view/ZmContactSplitView.js (hunk 2 above)
Comment (kalpeshhpatel) [inline WebRoot/js/zimbraMail/abook/view/ZmContactSplitView.js]: Done

Thanks

File: WebRoot/js/zimbraMail/abook/view/ZmGroupView.js (hunk 3 above)
Comment (kalpeshhpatel) [inline WebRoot/js/zimbraMail/abook/view/ZmGroupView.js]: Added comment. 

Thanks

File: WebRoot/skins/_base/base4/skin.css (hunk 4 above)
Comment (kalpeshhpatel) [inline WebRoot/skins/_base/base4/skin.css]: This CSS declaration has been removed and functionality is handled at template level as per review comment from @tarangkhandelwal 

Thanks

File: WebRoot/skins/_base/base4/skin.css (hunk 5 above)
Comment (kalpeshhpatel) [inline WebRoot/skins/_base/base4/skin.css]: Done. 

Thanks for pointing.

File: WebRoot/css/zm.css (hunk 6 above)
Comment (kalpeshhpatel) [inline WebRoot/css/zm.css]: Done. 

Thanks for suggestion.

File: WebRoot/js/zimbraMail/abook/view/ZmGroupView.js [hunk 7]
```diff
@@ -554,7 +554,7 @@ function(x, y, width, height) {
 	if(this._addNewField){
 		Dwt.setSize(this._addNewField, Dwt.DEFAULT, 100);
 	}
-	var memberListHeight = height - 50;
+	var memberListHeight = height - 150; //set member list height to totalContainerHeight - 150, to make sure that elements are of atleast enough required height
```
Comment (tarangkhandelwal) [inline WebRoot/js/zimbraMail/abook/view/ZmGroupView.js]: will there be a case when value of height would be <=150 ?

Comment (kalpeshhpatel) [inline WebRoot/js/zimbraMail/abook/view/ZmGroupView.js]: `height` is actually container height i.e height of full `Create Group/DL` page. So practically I don't see height to be less that 150px

**Pull request merged**